### PR TITLE
Add android package name support

### PR DESCRIPTION
**Pull request by coroiu**
Adds support for extra custom client data fields

BASED ON https://github.com/1Password/passkey-rs/pull/27

-----------------------

Bitwarden internal tracking:
[PM-8579] based on [PM-7720]

### CODE REVIEW

File: passkey-client/src/tests/mod.rs [hunk 1]
```diff
@@ -101,11 +104,79 @@ async fn create_and_authenticate() {
...
+    let credential_id = cred.raw_id;
+
+    let auth_options = webauthn::CredentialRequestOptions {
+        public_key: good_credential_request_options(credential_id),
+    };
+    let result = client
+        .authenticate(
+            &origin,
+            auth_options,
+            DefaultClientDataWithExtra(extra_data.clone()),
```
Comment (Progdrasil) [inline passkey-client/src/tests/mod.rs:162]: **Nit:** This clone is no longer necessary

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,15 @@
...
 - Changed: The `Client` no longer hardcodes the UV value sent to the `Authenticator` ([#22](https://github.com/1Password/passkey-rs/pull/22)).
+- Changed: The `Client` no longer hardcodes the RK value sent to the `Authenticator` ([#27](https://github.com/1Password/passkey-rs/pull/27)).
+- The client now supports additional user-defined properties in the client data, while also clarifying how the client
+handles client data and its hash.
+	- ⚠ BREAKING: Changed: `register` and `authenticate` take `ClientData<E>` instead of `Option<Vec<u8>>`.
+	- ⚠ BREAKING: Changed: Custom client data hashes are now specified using `DefaultClientDataWithCustomHash(Vec<u8>)` instead of
+		`Some(Vec<u8>)`.
+	- Added: Additional fields can be added to the client data using `DefaultClientDataWithExtra(ExtraData)`.
+- `CollectedClientData` is now generic and supports additional strongly typed fields.
+	- Changed: `CollectedClientData` has changed to `CollectedClientData<E = ()>`
```
Comment (Progdrasil) [inline CHANGELOG.md:27]: **Nit:** This is in `passkey-types` not `passkey-client`

File: passkey/src/lib.rs [hunk 3]
```diff
@@ -143,7 +143,7 @@
 //! };
 //!
 //! // Now create the credential.
-//! let my_webauthn_credential = my_client.register(&origin, request, None).await.unwrap();
+//! let my_webauthn_credential = my_client.register(&origin, request, DefaultClientData).await.unwrap();
```
Comment (Progdrasil) [inline passkey/src/lib.rs:146]: **Suggestion:** Mirror the example formatting

```suggestion
//! let my_webauthn_credential = my_client
//!     .register(&origin, request, DefaultClientData)
//!     .await
//!     .unwrap();
```

File: passkey-client/src/client_data.rs [hunk 4]
```diff
@@ -0,0 +1,54 @@
+use serde::Serialize;
+
+/// A trait describing how client data should be generated during a WebAuthn operation.
+pub trait ClientData<E: Serialize> {
+    /// Extra client data to be appended to the automatically generated client data.
+    fn extra_client_data(&self) -> E;
+
+    /// The hash of the client data to be used in the WebAuthn operation.
+    fn client_data_hash(&self) -> Option<Vec<u8>>;
+}
```
Comment (Progdrasil) [inline passkey-client/src/client_data.rs:10]: **Praise:** I really like this pattern

File: passkey-client/src/tests/mod.rs (hunk 1 above)
Comment (coroiu) [reply passkey-client/src/tests/mod.rs:158]: True, it's cloned further up so the original object is available here. Removed!

File: CHANGELOG.md (hunk 2 above)
Comment (coroiu) [reply CHANGELOG.md:37]: Moved!

File: passkey/src/lib.rs (hunk 3 above)
Comment (coroiu) [reply passkey/src/lib.rs]: Fixed!

File: passkey-client/src/client_data.rs (hunk 4 above)
Comment (coroiu) [reply passkey-client/src/client_data.rs:10]: Thank you! To be honest this is the part I thought had the biggest risk of having to be rewritten, glad you liked it!

Comment (Progdrasil) [reply passkey-client/src/client_data.rs:10]: Thats really impressive!

**Review by Progdrasil — APPROVED** [review]

**Pull request merged**